home work(lesson12)

Oleg-Korol opened the pull request:
hw12

Oleg-Korol committed:
home work(lesson12)

hw12

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+create_users_table =("""CREATE TABLE IF NOT EXISTS departaments(
+                 id INTEGER PRIMARY KEY AUTOINCREMENT,
+                 name str,
+                 manager_id ,
```
почему типа нет?

vladworldss commented on hw_12/Example_2.py:
```diff
@@ -0,0 +1,51 @@
+                 phone_number varchar(15),
+                 hire_date   varchar(10), 
+                 job_id        varchar(15),
+                 manager_id  INTEGER PRIMARY KEY AUTOINCREMENT,
```
получается, что каждый работник является менеджером?

vladworldss commented on hw_12/Example_3.py:
```diff
@@ -0,0 +1,47 @@
+
+#создаем таблицу
+create_users_table =("""CREATE TABLE IF NOT EXISTS jobs(
+                 title str PRIMARY KEY,
```
для первичных ключей обычно заводят искуственное поле - id или можешь назвать jobs_id

vladworldss commented on hw_12/Example_4.py:
```diff
@@ -0,0 +1,83 @@
+Sity=[]
+select_users = "SELECT * from location"
+users = execute_read_query(connection, select_users)
+for user in users:
```
зачем этот цикл, когда нужно все сделать на SQL. погугли, как получить отсортированный запрос

vladworldss reviewed (commented):
необходимо обработать комментарии

Oleg-Korol committed:
HW12 исправления

В 4 задании не смог применить ORDER и DISTINCT вместе ,так как сама  таблица не изменяется. Или нужно было создавать вторую таблицу?

Oleg-Korol committed:
+

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+
+connection = create_connection()
+#создаем таблицу
+create_users_table =("""CREATE TABLE IF NOT EXISTS departaments(
```
лучше давать имя, которое соотв имени таблицы. у тебя тут не users_table, а departments

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+execute_query(connection, create_users_table)
+
+#заполняем таблицу
+create_users =("""INSERT INTO departaments(name,manager_id,location_id) 
```
create_users_sql

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+execute_query(connection,create_users)
+
+#добавляем данные в таблицу
+def sql_insert(conection,val):
```
лучше давать более понятные имена, скажем create_department

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+#добавляем данные в таблицу
+def sql_insert(conection,val):
+    curs = conection.cursor()
+    curs.execute("""INSERT INTO departaments(name,manager_id,location_id) VALUES(?,?,?)""", val)
```
val лучше распилить конкретно на то кол-во, которое требуется, те дать норм имя аргументам

vladworldss commented on hw_12/Example_1.py:
```diff
@@ -0,0 +1,76 @@
+    result = curs.fetchone()
+    return result
+
+print("Количество сотрудников в отделе 'IT'=",*worker(connection))
```
давай откажемся от print? лучше писать явные проверки

vladworldss commented on hw_12/Example_2.py:
```diff
@@ -0,0 +1,52 @@
+
+#создаем таблицу
+create_users_table =("""CREATE TABLE IF NOT EXISTS Employees(
+                 Employs_id INTEGER PRIMARY KEY AUTOINCREMENT,
```
с малой буквы и в единств числе

vladworldss commented on hw_12/Example_2.py:
```diff
@@ -0,0 +1,52 @@
+#создаем таблицу
+create_users_table =("""CREATE TABLE IF NOT EXISTS Employees(
+                 Employs_id INTEGER PRIMARY KEY AUTOINCREMENT,
+                 first_name  varchar(10),
```
почему varchar?

vladworldss commented on hw_12/Example_2.py:
```diff
@@ -0,0 +1,52 @@
+                 phone_number varchar(15),
+                 hire_date   varchar(10), 
+                 job_id        varchar(15),
+                 manager_id VARCHAR(3) ,
```
у тебя id работника какой тип имеет?

vladworldss reviewed (commented):
- убрать принты, добавить норм проверки
- проанализировать применяемые типы, где может быть  NULL/нет
- добавить connection.close при завершении программы

Oleg-Korol committed:
Update Example_2.py
